### DISCUSSION TITLE
Assigning Journeys to Groups with Specific Waypoints and Exits

### DISCUSSION THREAD

**Discussion by K99-JDG** [PedestrianDynamics/jupedsim#1424] [Q&A] [answered]
I'm working on a second simulation where I need to assign agents to groups such that:

**Group 1 exits through the first exit via the first waypoint.
Group 2 exits through the second exit via the second waypoint.**

I've implemented the transitions in the journey correctly, but I’m encountering an issue when assigning these journeys to the groups. Specifically, I receive a RuntimeError: Unknown journey id when trying to add agents to the simulation.

I've attached a screenshot of the error!

Could someone point out what might be going wrong or suggest a better way to assign these journeys to the groups? Any help would be greatly appreciated!

<img width="696" alt="image" src="https://github.com/user-attachments/assets/0209a52f-fd26-4d67-812f-434bfe113563"
<img width="696" alt="스크린샷 2024-08-16 15 27 09" src="https://github.com/user-attachments/assets/41982b67-423e-42b2-8914-92991408e15b">

[PBTE_Case 2.ipynb.zip](https://github.com/user-attachments/files/16633727/PBTE_Case.2.ipynb.zip)

**Comment by chraibi** [comment, marked as answer]
Hi,
You work with an array of simulations `simulations[number]`, which is ok.
However, in this cell you add the journeys in  `simulation`, not in simulations[number]. This is why you get the error.
<img width="940" alt="Screenshot 2024-08-17 at 16 53 56" src="https://github.com/user-attachments/assets/ac2f2ded-245c-4cc2-9509-511248d70065">

Maybe you can wrap this cell in a function `init_journeys(simulation)` and return the journeys.


Maybe like this:
```
def init_journeys(simulation):
    waypoint_ids = [
        simulation.add_waypoint_stage([3.0125, 24.75], 0.45),
        simulation.add_waypoint_stage([3.0125, 4.95], 0.45),
    ]
    exit_ids = [
        simulation.add_exit_stage(exit_train),
        simulation.add_exit_stage(exit_platform),
    ]

    switch_id = simulation.add_waypoint_stage(switch_point, distance_to_switch)


    journey1 = jps.JourneyDescription([switch_id, waypoint_ids[0], exit_ids[0]])
    journey1.set_transition_for_stage(switch_id, jps.Transition.create_fixed_transition(waypoint_ids[0])),
    journey1.set_transition_for_stage(waypoint_ids[0], jps.Transition.create_fixed_transition(exit_ids[0])),
    journey_id1 = simulation.add_journey(journey1)

    journey2 = jps.JourneyDescription([switch_id, waypoint_ids[1], exit_ids[1]])
    journey2.set_transition_for_stage(switch_id, jps.Transition.create_fixed_transition(waypoint_ids[1])),
    journey2.set_transition_for_stage(waypoint_ids[1], jps.Transition.create_fixed_transition(exit_ids[1])),
    journey_id2 = simulation.add_journey(journey2)
    return journey_id1, journey_id2, switch_id

```

then later you do

```
for number, simulation in simulations.items():
    journey_id1, journey_id2, switch_id = init_journeys(simulation)
    group1 = set(
        [
            simulation.add_agent(
                jps.CollisionFreeSpeedModelAgentParameters(
                    position=position,
                    journey_id=journey_id1,
                    stage_id=switch_id,  
                )
            )
            for position in positions
        ]
    )
    .....
    ```

**Reply by K99-JDG** [reply]
It works perfectly! Thank you so much!
